### PR TITLE
sys/net/application_layer/nanocoap: add `nanocoap_sock_get_non()`

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description

We have NON-Confirmable versions for PUT, POST, and FETCH, but non for GET so far.
This PR adds it in a straight forward manner.

### Testing procedure

`tapsetup`
Two `native` instances of `tests/net/nanocoap_cli`.
One starts the server:
```
2024-09-05 15:48:28,576 # server start
2024-09-05 15:48:28,576 # starting server on port 5683
```

The other one GETs a resource:
```
> get_non coap://[fe80::a0ae:a2ff:fe2f:af16]/value
2024-09-05 15:48:32,316 # get_non coap://[fe80::a0ae:a2ff:fe2f:af16]/value
2024-09-05 15:48:32,318 # 00000000  30
```

### CODE REVIEW

File: sys/net/application_layer/nanocoap/sock.c [hunk 1]
```diff
@@ -390,7 +390,9 @@ static int _get_put_cb(void *arg, coap_pkt_t *pkt)
     return pkt->payload_len;
 }
 
-ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf, size_t len)
+ssize_t _sock_get(nanocoap_sock_t *sock, const char *path,
```
Comment (benpicco) [inline sys/net/application_layer/nanocoap/sock.c:393]:
```suggestion
static ssize_t _sock_get(nanocoap_sock_t *sock, const char *path,
```

File: sys/include/net/nanocoap_sock.h [hunk 2]
```diff
@@ -531,6 +531,20 @@ ssize_t nanocoap_sock_fetch_non(nanocoap_sock_t *sock, const char *path,
...
+/**
+ * @brief   Simple non-confirmable GET
+ *
+ * @param[in]   sock    socket to use for the request
+ * @param[in]   path    remote path and query
+ * @param[out]  response buffer for the response, may be NULL
+ * @param[in]   len_max length of @p response
+ *
+ * @returns     length of response payload on success
+ * @returns     <0 on error
```
Comment (Teufelchen1) [inline sys/include/net/nanocoap_sock.h:543]: What exactly is returned on error? (obviously not your fault here, `nanocoap_sock_get()` also does not specify this.)

Comment (Teufelchen1) [reply sys/include/net/nanocoap_sock.h:543]: Additionally, why not sort it up to said function?

Comment (fabian18) [reply sys/include/net/nanocoap_sock.h:543]: > Additionally, why not sort it up to said function?

All the `_non`  variants are further down.

Comment (Teufelchen1) [reply sys/include/net/nanocoap_sock.h]: Are you really sure? :sweat_smile: 

Isn't it: 
```
nanocoap_sock_Method()
nanocoap_sock_Method_non()
nanocoap_sock_Method_url()
```
With that, I would expect `.._get_non()` to be right after `.._get()`, near L390.

Comment (Teufelchen1) [reply sys/include/net/nanocoap_sock.h]: (I wouldn't block the PR for that but it really bugs me :bug:)

Comment (fabian18) [reply sys/include/net/nanocoap_sock.h]: Ah only in the header you mean. You got the pattern 👍

**Pull request merged**